lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -459,11 +463,11 @@ addIndices = mapM_ (`rawExecute` [])
 
 delegationToText :: W.WalletDelegation W.PoolId -> Maybe Text
 delegationToText W.NotDelegating = Nothing
-delegationToText (W.Delegating pool) = Just (W.getPoolId pool)
+delegationToText (W.Delegating pool) = Just $ toText pool
 
 delegationFromText :: Maybe Text -> W.WalletDelegation W.PoolId
 delegationFromText Nothing = W.NotDelegating
-delegationFromText (Just pool) = W.Delegating (W.PoolId pool)
+delegationFromText (Just pool) = W.Delegating (fromRight' $ fromText pool)
```

Comment:
Hey! This is a partial function :grimacing:! 
That only happens at the edge of the system with the database, right. I wonder though if we couldn't store a `PoolId` directly in the database and have the conversion done by `persistent` (cf the `.../DB/Sqlite/Types.hs` module) :thinking: ?